### PR TITLE
[BENCHMARKS] Remove XE-TLA references from fused_softmax

### LINKED ISSUE #6838: Fused softmax benchmark — replace xetla provider with sycl-tla or alternative
# Fused softmax benchmark — replace xetla provider with sycl-tla or alternative

**Parent**: #6724 — Switch XETLA benchmarks to SYCL_TLA analogues
**Complexity**: Low (given recommendation to simply drop xetla; no new kernel needed)
**Dependencies**: None

## Current state

`benchmarks/triton_kernels_benchmark/fused_softmax.py` uses xetla as a provider:

```python
from triton_kernels_benchmark import xetla_kernel, onednn_kernel
```

Providers: `triton`, `xetla`, `onednn`

The xetla softmax provider (lines 148-155):
- Calls shape-specific functions like `xetla_kernel.softmax_shape_4096_256`
- 7 shapes: `4096 x {256, 1024, 2048, 4096, 8192, 16384, 32768}` in BF16
- Has `assert_close` commented out (correctness not verified)

XeTLA C++ side (`xetla_kernel/softmax/`):
- Each shape is a compile-time templated configuration
- 7 pybind11 bindings in `python_main.cpp` (lines 206-219)

## SYCL-TLA documentation cross-reference

**Source**: [intel/sycl-tla docs](https://github.com/intel/sycl-tla/tree/main/media/docs/cpp)

### Confirmed: No softmax in SYCL-TLA
A thorough review of SYCL-TLA documentation and repository confirms:
- `functionality.md` lists only GEMM and convolution operations — no softmax
- `code_organization.md` shows no softmax directory or implementation
- `benchmarks/` directory contains only `gemm/` and `flash_attention/` subdirectories
- `applications/` directory contains only `dual_gemm/` and `flash_attention_v2/`
- The CUTLASS framework is fundamentally a matrix-multiply library; softmax is an element-wise reduction, not a natural fit for the CUTLASS tile/MMA abstraction

The flash attention implementation in SYCL-TLA does include an inline softmax epilogue (`FlashPrefillSoftmaxEpilogue` in `collective/`), but this is fused into the attention kernel — not exposed as a standalone operation.

## Key decision needed

**SYCL-TLA does NOT have and will not have a standalone softmax kernel.** Options:

### Option A: Add a SYCL-TLA softmax kernel
- Implement a CUTLASS-based softmax in `benchmarks/sycl_tla_kernel/softmax/`
- Expose via `sycl_tla_kernel.softmax(input, output, M, N)`
- Pro: Maintains the same benchmark comparison structure
- Con: Non-trivial implementation effort; softmax is not a standard CUTLASS operation. SYCL-TLA's architecture (tile-based GEMM with DPAS atoms) is not designed for row-wise reductions. Would require writing a custom SYCL kernel outside the CUTLASS framework, defeating the purpose.

### Option B: Drop xetla and keep only triton + onednn
- Remove the xetla provider entirely
- The benchmark still compares Triton vs OneDNN (PyTorch native softmax)
- Pro: Simple, onednn is a strong baseline
- Con: Loses the XeTLA "handwritten SYCL kernel" comparison point

### Option C: Replace xetla with PyTorch native (`torch.softmax`)
- Already available as commented-out `torch-native` and `torch-jit` providers
- Uncomment and use `torch.softmax` as the reference baseline
- Pro: Zero new code; PyTorch softmax calls oneDNN under the hood on XPU anyway
- Con: May be redundant with the existing onednn provider

**Recommendation**: Option B — drop xetla, keep triton + onednn. The onednn provider already provides a strong native baseline. SYCL-TLA is architecturally unsuitable for standalone softmax, so there is no meaningful replacement.

### 1. Remove xetla from `fused_softmax.py`
- Remove `xetla_kernel` from import
- Remove `"xetla": "XeTLA"` from `supported_providers` dict
- Remove the `elif provider == "xetla":` block (lines 148-155)
- Update comments referencing xetla warmup counts

### 2. Update benchmark metadata
- Fix `styles` list to match remaining providers
- Update warmup comments

## Files to modify
- `benchmarks/triton_kernels_benchmark/fused_softmax.py`

## Testing
- Run: `python benchmarks/triton_kernels_benchmark/fused_softmax.py`
- Verify triton and onednn providers work for all 7 shapes

### PR DESCRIPTION
**Pull request by slawblauciak**

### CODE REVIEW

File: benchmarks/triton_kernels_benchmark/fused_softmax.py [hunk 1]
```diff
@@ -123,14 +122,12 @@ def get_benchmark(providers_filter: Optional[list[str]] = None):
             line_arg="provider",  # argument name whose value corresponds to a different line in the plot
             line_vals=list(providers.keys()),  # possible values for `line_arg``
             line_names=list(providers.values()),  # label name for the lines
-            styles=[("blue", "-"), ("green", "-"), ("green", "--"), ("black", ":")],  # line styles
```
Comment (anmyachev) [inline benchmarks/triton_kernels_benchmark/fused_softmax.py:126]: Why were 2 styles removed?

File: benchmarks/triton_kernels_benchmark/fused_softmax.py [hunk 2]
```diff
@@ -123,14 +122,12 @@ def get_benchmark(providers_filter: Optional[list[str]] = None):
...
+            styles=[("blue", "-"), ("green", "-")],  # line styles
             ylabel=["GB/s", "TFlops"],  # label name for the y-axis
             plot_name="softmax-performance",  # name for the plot. Used also as a file name for saving the plot.
             args={"M": 4096},  # values for function arguments not in `x_names` and `y_name`
         ))
     def benchmark(M, N, provider):
-        # Maximum across torch-native=10, triton=800, torch-jit=10, xetla=100, onednn=800
-        # For onednn more warmup very slowly makes performance worse
```
Comment (anmyachev) [inline benchmarks/triton_kernels_benchmark/fused_softmax.py:133]: Let's leave the comment.

**Comment by anmyachev** [comment]
@slawblauciak you also need to remove `--compiler xetla` option in `triton-benchmarks*.yml` files.

File: benchmarks/triton_kernels_benchmark/fused_softmax.py (hunk 1 above)
Comment (slawblauciak) [reply benchmarks/triton_kernels_benchmark/fused_softmax.py:126]: There's only 2 providers

Comment (anmyachev) [reply benchmarks/triton_kernels_benchmark/fused_softmax.py:126]: Ok, make sure the styles for existing ones haven't changed.

**Review by whitneywhtsang — APPROVED** [review]

**Comment by anmyachev** [comment]
FYI @slawblauciak there is merge conflict

**Review by whitneywhtsang — APPROVED** [review]

**Review by anmyachev — APPROVED** [review]
